holdings_maintenance/pds4/pds4infoshelf.py

```
@@ -0,0 +1,925 @@
+    # Read info shelf file
+    shelf_infodict = load_infodict(pdsdir, logger=logger)
+
+    # Generate info
+    (dir_infodict, latest_modtime) = generate_infodict(pdsdir, selection,
+                                                       logger=logger)
+    latest_iso = latest_modtime.replace(' ', 'T')
+    latest_datetime = datetime.datetime.fromisoformat(latest_iso)
+
+    # For a single selection, use the old information
+    if selection:
+        key = list(dir_infodict.keys())[0]
```

Ruff (0.15.2)

[warning] Prefer `next(iter(dir_infodict.keys()))` over single element slice

Replace with `next(iter(dir_infodict.keys()))`

(RUF015)